common/perf-test-core/src/main/java/com/azure/perf/test/core/ApiPerfTestBase.java

```diff
@@ -80,30 +67,81 @@ public ApiPerfTestBase(TOptions options) {
         }
     }
 
+    private static HttpClient createHttpClient(PerfStressOptions options) {
+        PerfStressOptions.HttpClientType httpClientType = options.getHttpClient();
+        switch (httpClientType) {
+            case NETTY:
+                if (options.isInsecure()) {
+                    try {
+                        SslContext sslContext = SslContextBuilder.forClient()
+                            .trustManager(InsecureTrustManagerFactory.INSTANCE)
+                            .build();
+
+                        reactor.netty.http.client.HttpClient nettyHttpClient =
+                            reactor.netty.http.client.HttpClient.create()
+                            .secure(sslContextSpec -> sslContextSpec.sslContext(sslContext));
+
+                        return new NettyAsyncHttpClientBuilder(nettyHttpClient).build();
+                    } catch (SSLException e) {
+                        throw new IllegalStateException(e);
+                    }
+                } else {
+                    return new NettyAsyncHttpClientProvider().createInstance();
+                }
+            case OKHTTP:
+                if (options.isInsecure()) {
+                    try {
+                        SSLContext sslContext = SSLContext.getInstance("SSL");
+                        sslContext.init(
+                            null, InsecureTrustManagerFactory.INSTANCE.getTrustManagers(), new SecureRandom());
+                        OkHttpClient okHttpClient = new OkHttpClient.Builder()
+                            .sslSocketFactory(sslContext.getSocketFactory(),
+                                (X509TrustManager) InsecureTrustManagerFactory.INSTANCE.getTrustManagers()[0])
+                            .build();
+                        return new OkHttpAsyncHttpClientBuilder(okHttpClient).build();
+                    } catch (NoSuchAlgorithmException | KeyManagementException e) {
+                        throw new IllegalStateException(e);
+                    }
+                } else {
+                    return new OkHttpAsyncClientProvider().createInstance();
+                }
+            default:
+                throw new IllegalArgumentException("Unsupported http client " + httpClientType);
+        }
+    }
+
+    private static reactor.netty.http.client.HttpClient createRecordPlaybackClient(PerfStressOptions options) {
```

Comment:
Should the record and playback HttpClient also support Netty and OkHttp?

Reply:
Not really, this client is just used for sideline communication between test proxy and core perf classes for "kind of RPC over REST". Doesn't affect actual SDK clients at all, so there's not value in doing that.